core/src/executor/evaluate/evaluated.rs

```diff
@@ -290,27 +290,39 @@ impl<'a> Evaluated<'a> {
         Ok(evaluated)
     }
 
-    pub fn like(&self, other: Evaluated<'a>, case_sensitive: bool) -> Result<Evaluated<'a>> {
+    pub fn like(
+        &self,
+        other: Evaluated<'a>,
+        case_sensitive: bool,
+        escape_char: char,
+    ) -> Result<Evaluated<'a>> {
         let evaluated = match (self, other) {
             (Evaluated::Literal(l), Evaluated::Literal(r)) => {
                 Evaluated::Literal(l.like(&r, case_sensitive)?)
```

Comment:
1. Escape should be implemented at Literal::like as well.
2. Then, you can write test case on `SELECT clause` with simple comparison like below.
3. We created new `test!` macro which includes `name` field to help understanding. you can just copy this test cases to `like_ilike.rs`.
```rs
    test! {
        name: "`_` should match any single character",
        sql: r#"SELECT 'GLUE+SQL' LIKE 'GLUE_SQL'"#,
        expected: Ok(select!(column1 Bool; true))
    };

    test! {
        name: r#"`\` should escape `_` to normal character - false"#,
        sql: r#"SELECT 'GLUE+SQL' LIKE 'GLUE\_SQL'"#,
        expected: Ok(select!(column1 Bool; false))
    };

    test! {
        name: r#"`\` should escape `_` to normal character - true"#,
        sql: r#"SELECT 'GLUE_SQL' LIKE 'GLUE\_SQL'"#,
        expected: Ok(select!(column1 Bool; true))
    };

    test! {
        name: "`%` should match any sequence of zero or more characters",
        sql: r#"SELECT 'GLUE+-SQL' LIKE 'GLUE%SQL'"#,
        expected: Ok(select!(column1 Bool; true))
    };

    test! {
        name: r#"`\` should escape `%` to normal character - false"#,
        sql: r#"SELECT 'GLUE+-SQL' LIKE 'GLUE\%SQL'"#,
        expected: Ok(select!(column1 Bool; false))
    };

    test! {
        name: r#"`\` should escape `%` to normal character - true"#,
        sql: r#"SELECT 'GLUE%SQL' LIKE 'GLUE\%SQL'"#,
        expected: Ok(select!(column1 Bool; true))
    };

    test! {
        name: r#"`\` should escape `\` to normal character"#,
        sql: r#"SELECT 'GLUE\SQL' LIKE 'GLUE\\SQL'"#,
        expected: Ok(select!(column1 Bool; true))
    };

    test! {
        name: r#"`\` should escape `'` to normal character"#,
        sql: r#"SELECT 'GLUE''SQL' LIKE 'GLUE\'SQL'"#,
        expected: Ok(select!(column1 Bool; true))
    };

    test! {
        name: r#"ESCAPE clause should set escape character (default: `\\`)"#,
        sql: r#"SELECT 'GLUE_SQL' LIKE 'GLUE\_SQL' ESCAPE '\\'"#,
        expected: Ok(select!(column1 Bool; true))
    };

    test! {
        name: r#"ESCAPE clause should set escape character to `#`"#,
        sql: r#"SELECT 'GLUE_SQL' LIKE 'GLUE#_SQL' ESCAPE '#'"#,
        expected: Ok(select!(column1 Bool; true))
    };

    test! {
        name: r#"`#` should escape `#` to normal character"#,
        sql: r#"SELECT 'GLUE#SQL' LIKE 'GLUE##SQL' ESCAPE '#'"#,
        expected: Ok(select!(column1 Bool; true))
    };
```

Reply:
In `Like clause of SQL`, `_` is special character which means `match any single character`.
But in second test (`SELECT 'GLUE+SQL' LIKE 'GLUE\_SQL'`), `_` was escaped by `\` so that it became just normal character.
Then, the test result should be `false` as `+` is different with normal character `_`

Sixth test (`SELECT 'GLUE%SQL' LIKE 'GLUE\%SQL'`) has similar goal.
`%` is special character which means `match any sequence of zero or more characters`
But `%` was escaped by `\` so that it became just normal character.
Then, the test result should be `true` as `%` is equal to normal character `%`

That was why i wanted to test `escaping` all the special characters(`_`, `%`, `\`, `'`) in `Like clause of SQL`

Reply:
The reason why CI test was failed is that the test cases i gave you have wrong column name.
It looks like each sql should use explicit alias

```diff
test! {
        name: "`_` should match any single character",
-        sql: r#"SELECT 'GLUE+SQL' LIKE 'GLUE_SQL'"#,
+        sql: r#"SELECT 'GLUE+SQL' LIKE 'GLUE_SQL' AS column1"#,
        expected: Ok(select!(column1 Bool; true))
    };
```

Reply:
@devgony the changes pushed but in this case : `SELECT 'GLUE\SQL' LIKE 'GLUE\\SQL'` . the regex should be `^GLUE\SQL$` and match string is `"GLUE\\SQL"` and not match . I am new regex :) hope you can solve it and commit the learn about it . thanks .